MonoGame.Framework.Content.Pipeline/Audio/AudioContent.cs

```diff
@@ -134,229 +114,188 @@ int QualityToBitRate(ConversionQuality quality)
+        /// the Data property.
+        /// </param>
+        public void ConvertFormat(ConversionFormat formatType, ConversionQuality quality, string saveToFile)
         {
             if (disposed)
                 throw new ObjectDisposedException("AudioContent");
 
-
-        switch (formatType)
+            var temporarySource = Path.GetTempFileName();
+            var temporaryOutput = Path.GetTempFileName();
+            try
             {
-                case ConversionFormat.Adpcm:
-#if WINDOWS
-                    ConvertWav(new AdpcmWaveFormat(QualityToSampleRate(quality), format.ChannelCount));
-                    break;
-#else
-                    throw new NotSupportedException("Adpcm encoding supported on Windows only");
-#endif
-
-                case ConversionFormat.Pcm:
-#if WINDOWS
-                    ConvertWav(new WaveFormat(QualityToSampleRate(quality), format.ChannelCount));
-                    break;
-#elif LINUX
-                    // TODO Do the conversion for Linux platform
-                    throw new NotSupportedException("Pcm has not been implemented on this platform");
-#else //MONOMAC
-                    targetFileName = Guid.NewGuid().ToString() + ".wav";
-                    if (!ConvertAudio.Convert(fileName, targetFileName, AudioFormatType.LinearPCM, MonoMac.AudioToolbox.AudioFileType.WAVE, quality)) {
-                        throw new InvalidDataException("Failed to convert to PCM");
-                    }
-                    Read(targetFileName);
-                    if (File.Exists(targetFileName))
-                        File.Delete(targetFileName);
-                    break;
-#endif
+                using (var fs = new FileStream(temporarySource, FileMode.Create, FileAccess.Write))
+                {
+                    var dataBytes = this.data.ToArray();
+                    fs.Write(dataBytes, 0, dataBytes.Length);
+                }
 
-                case ConversionFormat.WindowsMedia:
-#if WINDOWS
-                    reader.Position = 0;
-                    MediaFoundationEncoder.EncodeToWma(reader, targetFileName, QualityToBitRate(quality));
-                    break;
-#else
-                    throw new NotSupportedException("WindowsMedia encoding supported on Windows only");
-#endif
+                string ffmpegCodecName, ffmpegMuxerName;
+                int format;
+                switch (formatType)
+                {
+                    case ConversionFormat.Adpcm:
+                        // ADPCM Microsoft 
+                        ffmpegCodecName = "adpcm_ms";
+                        ffmpegMuxerName = "wav";
+                        format = 0x0002; /* WAVE_FORMAT_ADPCM */
+                        break;
+                    case ConversionFormat.Pcm:
+                        // PCM signed 16-bit little-endian
+                        ffmpegCodecName = "pcm_s16le";
+                        ffmpegMuxerName = "s16le";
```

Comment:
The raw data format is used because it is that raw data that is put in the XNB file.